fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2110,6 +2270,55 @@ public Range getUnbuiltRange() {
             }
             return indexStateMap;
         });
+        if (timer != null) {
+            result = timer.instrument(FDBStoreTimer.Events.LOAD_RECORD_STORE_INDEX_STATE, result, context.getExecutor());
+        }
+        return result;
+    }
+
+    @Nonnull
+    private CompletableFuture<Map<String, IndexMetaDataProto.IndexMetaData>> loadIndexMetaDataAsync(@Nonnull IsolationLevel isolationLevel) {
+        Subspace indexMetaDataSubspaceSpace = indexMetaDataSubspace();
+        KeyValueCursor cursor = KeyValueCursor.Builder.withSubspace(indexMetaDataSubspaceSpace)
+                .setContext(getContext())
+                .setRange(TupleRange.ALL)
+                .setContinuation(null)
+                .setScanProperties(new ScanProperties(ExecuteProperties.newBuilder().setIsolationLevel(isolationLevel).build()))
+                .build();
+        FDBStoreTimer timer = getTimer();
+        CompletableFuture<Map<String, IndexMetaDataProto.IndexMetaData>> result = cursor.asList().thenApply(list -> {
+            Map<String, IndexMetaDataProto.IndexMetaData> indexMetaDataMap;
+            if (list.isEmpty()) {
+                indexMetaDataMap = Collections.emptyMap();
+            } else {
+                ImmutableMap.Builder<String, IndexMetaDataProto.IndexMetaData> indexMetaDataMapBuilder = ImmutableMap.builder();
+
+                for (KeyValue kv : list) {
+                    try {
+                        String indexName = getRecordMetaData().getIndex(indexMetaDataSubspaceSpace.unpack(kv.getKey()).get(0)).getName();
+                        indexMetaDataMapBuilder.put(indexName, IndexMetaDataProto.IndexMetaData.parseFrom(Tuple.fromBytes(kv.getValue()).getBytes(0)));
+                    } catch (MetaDataException ex) {
+                        // The index is somehow undefined. Just pass.
+                        // throw new RecordCoreStorageException("Loading undefined index ", ex);
+
+                        // TODO: Question for the reviewers: If we throw error (like how I used to do above), we could end up in
+                        //  an unrecoverable situation that the only way out of it is wiping the index meta-data.
+                        //  Shall we just skip this index? Test FDBRecordStore.open() is one example of this case.
```

Comment:
Yeah, I'm not in favor of an exception here--or, frankly, even a WARN, because it could get really annoying.  I think, though, probably a DEBUG log message at least would be appropriate.